android/src/main/java/com/reactlibrary/RNReactNativeDocViewerModule.java

```diff
@@ -325,7 +325,10 @@ protected void onPostExecute(File result) {
             Context context = getCurrentActivity();
            String mimeType;
             // mime type of file data
-            if (fileName != null && fileType != null) {
+            if (fileType != null) {
+                // If file type is already specified, should just take the mimeType from it
+                mimeType = MimeTypeMap.getSingleton().getMimeTypeFromExtension(fileType);
+            } else if (fileName != null && fileType != null) {
```

Comment:
You could also remove the first `else if` statement, since it would never execute. If `fileType` is null (so the `if` statement doesn't execute), then the first `else if` statement will also be false (because it also checks for `fileType != null`)

Reply:
That is true. Thanks for pointing out!